### PR TITLE
feat: Add possibility to not enforce default warehouse / role / namespace

### PR DESCRIPTION
**Pull request by dgniewek**
Add possibility to create user without enforcing:
* `default_warehouse` 
 * `default_role`
 * `default_namespace`

The above configurations will be added to terraform's logic `ignore_differences` when `enforce_defaults` will be set to `false`

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -112,7 +112,13 @@ variable "must_change_password" {
 }
 
 variable "grant_default_roles" {
-  description = "Whether to grant default_role and default_secondary_roles to Snowflake User"
+  description = "Whether to grant default_role to Snowflake User"
+  type        = bool
+  default     = true
+}
+
+variable "enforce_defaults" {
```
Comment (jakubigla) [inline variables.tf:120]: perhaps `ignore_changes_on_defaults` is more descriptive?

File: outputs.tf [hunk 2]
```diff
@@ -1,51 +1,51 @@
 output "default_namespace" {
   description = "Specifies the namespace (database only or database and schema) that is active by default for the user's session upon login"
-  value       = one(snowflake_user.this[*].default_namespace)
+  value       = var.enforce_defaults ? one(snowflake_user.this[*].default_namespace) : one(snowflake_user.defaults_not_enforced[*].default_namespace)
```
Comment (jakubigla) [inline outputs.tf:3]: we could have it cleaner with:
```
locals {
  snowflake_user = var.enforce_defaults ? snowflake_user.this : snowflake_user.defaults_not_enforced
}

output "default_namespace" {
  //...
  value = one(local.snowflake_user[*].default_namespace)
}

```

File: variables.tf (hunk 1 above)
Comment (dgniewek) [reply variables.tf:120]: It was my orginal idea, but `enforce_defaults` was just shorter... But I can rollback

**Review by jakubigla — APPROVED** [review]
LGTM

**Pull request merged**